examples/decoding/plot_ssd_spatial_filters.py

```diff
@@ -0,0 +1,435 @@
+                                  fir_design='firwin'),
+          filt_params_noise=dict(l_freq=freqs_noise[0], h_freq=freqs_noise[1],
+                                 l_trans_bandwidth=1, h_trans_bandwidth=1,
+                                 fir_design='firwin'),
+          sampling_freq=raw.info['sfreq'])
+
+# prepare data
+X = raw.copy().pick(picks_raw).get_data()
+
+ssd.fit(X)
+
+
+# Let's investigate spatila filter with max power ratio.
+# We will first inspect the topographies.
+# According to Nikulin et al 2011 this is done.
+# by either inverting the filters (W^{-1}) or by multiplying the noise
+# cov with the filters Eq. (22) (C_n W)^t.
+# We rely on the inversion approach here.
+
+plt.close('all')
+pattern = mne.EvokedArray(data=ssd.patterns_[:4].T,
+                          info=mne.pick_info(raw.info, picks_raw))
+pattern.plot_topomap(units=dict(mag='A.U.'), time_format='')
+
+# The topographies suggest that we picked up a parietal alpha generator.
+
+# transform
+ssd_sources = ssd.transform(X)
+
+# get psd of SSD-filtered signals
+psd, freqs = mne.time_frequency.psd_array_welch(
+    ssd_sources, sfreq=sf, n_fft=4096)
+
+# get spec_ratio information
+spec_ratio = ssd.spec_ratio
+sorter = ssd.sorter_spec
+
+# plot spectral ratio (see Eq. 24 in Nikulin 2011)
+plt.figure()
+plt.plot(spec_ratio, color='black')
+plt.plot(spec_ratio[sorter], color='orange', label='sorted eigenvalues')
+plt.xlabel("Eigenvalue Index")
+plt.ylabel(r"Spectral Ratio $\frac{P_f}{P_{sf}}$")
+plt.legend()
+plt.axhline(1, linestyle='--')
+
+# Let's also look at tbe power spectrum of that source and compare it to
+# to the power spectrum of the source with lowest SNR.
+
+
+below50 = freq_mask(freqs, 0, 50)
+# for highlighting the freq. band of interest
+bandfilt = freq_mask(freqs, freqs_sig[0], freqs_sig[1])
+plt.figure()
+plt.loglog(freqs[below50], psd[0, below50], label='max SNR')
+plt.loglog(freqs[below50], psd[-1, below50], label='min SNR')
+plt.loglog(freqs[below50], psd[:, below50].mean(axis=0), label='mean')
+plt.fill_between(freqs[bandfilt], 0, 10000, color='green', alpha=0.15)
+plt.ylim(0, 10000)
+plt.xlabel("log(frequency)")
+plt.ylabel("log(power)")
+plt.legend()
+
+# We can clearly see that the selected component enjoys an SNR that is
+# way above the average powe spectrum.
+
+# Epoched data
+# Although we suggest to use this method before epoching, there might be some
+# situations in which data can only be treated by chunks
+
+# Build epochs as sliding windows over the continuous raw file
+events = mne.make_fixed_length_events(raw, id=1, duration=2.0, overlap=0.0)
+
+# Epoch length is 1 second
+epochs = Epochs(raw, events, tmin=0., tmax=1, baseline=None,
+                picks=picks_raw,
+                detrend=1, decim=5, preload=True)
+
+# Build epochs as sliding windows over the continuous raw file
+events = mne.make_fixed_length_events(raw, id=1, duration=5.0, overlap=0.0)
+
+# Epoch length is 1 second
```

Comment:
here now seems to be 5 sec

Reply:
yeah heck it was duplicated somehow!
I kept 5 seconds to make sure filter length is the same when doing Raw VS Epochs.
Like this we can compare.